src/react/components/pages/homepage/homePage.tsx

```diff
@@ -115,9 +118,20 @@ export default class HomePage extends React.Component<IHomepageProps> {
         this.cloudFilePicker.current.open();
     }
 
-    private onProjectFileUpload = async (e, projectJson) => {
-        const project: IProject = JSON.parse(projectJson);
-        await this.loadSelectedProject(project);
+    private onProjectFileUpload = async (e, project) => {
+        let projectJson: IProject;
+        try {
+            projectJson = JSON.parse(project);
+        } catch (error) {
+            this.props.appErrorActions.showError({
+                title: "Homepage has an error",
+                message: "File is not valid json",
+            });
+
+            return;
+        }
+
+        await this.loadSelectedProject(projectJson);
```

Comment:
You can move this `await this.loadSelectedProject(projectJson);` into the `try` block which will then avoid you having to have the return statement in your `catch` block

Reply:
that mean we could have an error that is coming from loadSelectedProject and not just from json parsing, which is fine but the error message would have to be more generic